lib/galaxy/tools/__init__.py

```diff
@@ -107,6 +108,32 @@ def __init__( self, config_filenames, tool_root_dir, app ):
             tool_root_dir=tool_root_dir,
             app=app,
         )
+        self._view = views.DependencyResolversView(app)
+
+    def all_tools_requirements(self):
+        reqs = []
+        for desc, tool in self.tools():
+            for requirement in tool.requirements:
+                if requirement.type == 'package':
+                    reqs.append(json.dumps(requirement.to_dict(), sort_keys=True))
+        reqs = [json.loads(req) for req in set(reqs)]
+        return reqs
+
+    def tool_requirements(self, tool):
+        reqs = [req.to_dict() for req in tool.requirements if req.type == 'package']
+        return reqs
+
+    def tool_requirements_status(self, id):
+        tool = self._tools_by_id[id]
+        requirements = self.tool_requirements(tool)
+        installed_requirements = self._view.manager_requirements()
+        result = []
+        for req in requirements:
+            for ireq in installed_requirements:
+                if req == ireq['requirement']:
+                    req['status'] = "installed through %i" % ireq['index']
+                    result.append(req)
+        return result
```

Comment:
I think these 2 methods (`tool_requirements()` and `tool_requirements_status()`) should probably be moved to class `Tool`.